### ISSUE TITLE
CI job to deploy User docs to Syndesis website

### ISSUE THREAD

**Issue by kahboom** [syndesisio/syndesis.io#99]
**tl;dr**
We need to create a CI job that deploys User docs to the Syndesis website (`gh-pages` branch via the `master` branch build).

**Background**
Per conversation w/ @KurtStam and @zregvart , it would be beneficial to keep the User docs in the main [Syndesis app repository](https://github.com/syndesisio/syndesis/tree/master/doc), as they currently are, for a few reasons:

- This is what we currently have set up (less work)
- Avoid disrupting the Docs team's existing workflow
- Keeping the user documentation close to the code base fosters community involvement
- Easier alignment with versioning (similar to how Apicurio has set up [here](https://apicurio-studio.readme.io/docs) with the version dropdown)

So, the setup will be as follows:

- Developer Handbook and other docs (contribution guidelines, quickstarts, etc.) will exist in the [Syndesis website repository](https://github.com/syndesisio/syndesis.io).
- Community User docs will remain in the main [Syndesis app repository](https://github.com/syndesisio/syndesis/tree/master/doc). A CI build will deploy the docs to the Syndesis website.

Both types of documentation, on building, will end up on the [Syndesis community website](https://syndesis.io/). This setup will also be documented on the website. 😂
